### PR TITLE
Allow arbitrary inter-document callback ordering

### LINKED ISSUE #29: It's odd to half-define inter-document callback ordering, but not fully define it
The fact that inter-document ordering is half-defined and half-arbitrary is a bit weird.  Specifically, if I have two documents, A and B, and each schedules two idle callbacks (A1, A2, B1, B2) then I think per this spec the only allowable execution orders are "A1, B1, A2, B2" and "B1, A1, B2, A2", but either one of those two is ok.  That's assuming those 4 idle callbacks fit inside a single idle period.  If the idle period ends after the first two have run, then "A1, B1, B2, A2" and "B1, A1, A2, B2" also become possible orders, and if the idle period ends after every callback then the other orderings that keep A1 before A2 and B1 before B2 also become possible. 

Given that, I'm not sure why a UA is required to interleave things in a particular way inside a single idle period; it seems like an accident of the way the processing model is defined, and allows direct observation of when idle periods start/end in a UA, and I think it would be better to not overconstrain things in that way.

### PR DESCRIPTION
**Pull request by igrigorik**
Closes #29.

### CODE REVIEW

**Comment by igrigorik** [comment]
@bzbarsky @rmcilroy as proposed in https://github.com/w3c/requestidlecallback/issues/29#issuecomment-214880546, please take a quick look.

File: index.html [hunk 1]
```diff
@@ -391,14 +391,17 @@
 </div>
 <p>The <dfn>invoke idle callbacks algorithm</dfn>:</p>
 <ol>
+<li>Let <var>windows</var> be the <code>Window</code> object's list of <a href="https://www.w3.org/TR/html5/browsers.html#browsing-context">browsing context</a>'s <code>WindowProxy</code> objects that have a non-empty <a>list of runnable idle callbacks</a> (<var>runlist</var>).</li>
```
Comment (bzbarsky) [inline index.html]: A Window doesn't have any sort of list of WindowProxy objects associated with it, except its child windows, I suppose.  I'm not sure what this is trying to say.

Comment (igrigorik) [inline index.html]: Hmm, I should have mentioned that this pull depends on https://github.com/w3c/requestidlecallback/pull/38 - related discussion in https://github.com/w3c/requestidlecallback/issues/25. Can you please take a look at that?

**Comment by bzbarsky** [comment]
I commented there; I don't see why WindowProxy is involved at all.

**Comment by igrigorik** [comment]
@bzbarsky rebased against updated #38. Can you take another look?

**Comment by bzbarsky** [comment]
Looks reasonable, I think.  Reading HTML diffs is no fun.  ;)

**Comment by igrigorik** [comment]
@bzbarsky thanks, merging! 

Re, diffs: yeah, and that docs needs some formatting love. On my list..

**Pull request merged**